controllers/utils/common.go

```diff
@@ -56,5 +65,23 @@ func NewSafeResourceName(name, suffix string, maxLength, spareLength int) string
 	} else {
 		base = name
 	}
-	return fmt.Sprintf("%s-%s", base, suffix)
+
+	// Make sure base ends in '-' or an alphanumerical character.
+	for !regexp.MustCompile(`^[a-zA-Z0-9-]*$`).MatchString(base[len(base)-1:]) {
+		base = base[:len(base)-1]
+	}
+
+	newSafeResourceName := fmt.Sprintf("%s-%s", base, suffix)
+	// Validate the newSafeResourceName matches the required regex as per
+	// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+	match, err := regexp.MatchString(RFC1123MetadataRegex, newSafeResourceName)
```

Comment:
Is this really necessary? We pretty much know base is going to be a well formed name with this change. How can adding the suffix make it not match?

Reply:
Sure, please do that if you don't mind. In general, it's kind of bad to include code that can't be tested. The test coverage number will suffer a little :)